### ISSUE TITLE
Saving project in a Scratch embedded into Django

### ISSUE THREAD

**Issue by martingallardo23** [LLK/scratch-gui#8918]
This is surely a beginners questions, but I really tried to sort it out and couldn't figure out how to do it. I have a Django project where I embedded a local instance of Scratch by cloning this repo, building it, and copying the build files into the static folder in the Django project. The embedding works fine but I would like to be able to program that after the user hits a "submit" button it triggers a save project into .sb3 format. Is there any way to achieve that? I know that the user could manually save it by going to files -> save to your computer; but that doesn't work for my purposes (I need to save it to a database). 

Thanks!

**Comment by Racecar-Roaster** [comment]
It could possible to programmatically save a Scratch project as an `.sb3` file in a Django project. You can use the `subprocess` module to execute the `save` command in the Scratch GUI, and then save the resulting `.sb3` file to your database.

While it is in theory _possible_ to programmatically save a Scratch project as an `.sb3` file using the `subprocess` module, it is important to note that this involves interacting with the Scratch GUI, which could be vulnerable to race conditions or other issues that could compromise the integrity of the resulting `.sb3` file. Additionally, Scratch is primarily designed for educational purposes, and may not be suitable for use in secure or critical applications. Before implementing this solution, it is recommended to carefully consider the potential risks and design a robust system that can handle any unexpected issues that may arise.

**Comment by CST1229** [comment]
> Just so you know this may not be correct but it is reccomended to test it out since I got that help from someone else.

![](https://user-images.githubusercontent.com/68464103/235691587-8c20d5b8-3e83-4954-9760-84a621af2288.png)

**Comment by furkancgrtkn** [comment]
Hi, did you solve it ?

**Comment by martingallardo23** [comment]
Yes, but I'm not much of a programmer and knew nothing about React so there's surely cleaner ways to do it. What I did was basically add a function in Scratch that takes the current project and sends a POST request with the project as payload to an API in my Django site, which then takes it and saves it to an S3 bucket. Since the Scratch page is embedded in the Django site, I added a button in my site that sends a message to the Scratch iframe to trigger the function.
